fastdtw/_fastdtw.pyx

```diff
@@ -0,0 +1,166 @@
+#!python
+#cython: boundscheck=False, cdivision=True, wraparound=False
+
+from __future__ import absolute_import, division
+
+from libcpp.vector cimport vector  # noqa
+from libcpp.map cimport map
```

Comment:
C++11 has `unordered_map` and `unordered_set`. They are more resemblant of Python's set and dict as they provide O(1) access (as opposed to `map` and `set` which are treeish with O(log(N))).

You may have tried it and got an error. The problem is that C++ `std::hash()` isn't defined for `pair` objects. What you can do is hack the `__get_key()` to return an integer value, like this:
```cython
ctypedef long long key_t  # 64 bits should be enough for everyone

cdef inline key_t __get_key(int i, int j):
    return (<key_t>i << 32) + j
```
and then you have:
```cython
cdef unordered_map[key_t, CostEntry] D
```
which should work.